### DISCUSSION TITLE
pg_dump from the Supabase documentation doesn't work. What am I doing wrong?

### DISCUSSION THREAD

**Discussion by RikNieuwoudt** [supabase/supabase#3464] [Questions] [answered]
I copied the pg_dump CLI from https://supabase.io/docs/guides/database#migrating-between-projects
`pg_dump -h db.PROJECT_REF.supabase.co -U postgres --data-only -n public -n auth > supabase_data.sql`

But I'm getting the following errors
`pg_dump: error: query failed: ERROR:  permission denied for sequence refresh_tokens_id_seq
pg_dump: error: query was: SELECT last_value, is_called FROM auth.refresh_tokens_id_seq`

I have been stuck on trying to backup my db from Supabase for weeks already - please help?

**Comment by soedirgo** [comment, marked as answer]
Hey @RikNieuwoudt, we're aware of the issue and are working on a fix. For now, you can run the following in the SQL editor and try `pg_dump`ing again:
```sql
GRANT ALL ON ALL SEQUENCES IN SCHEMA auth TO postgres; 
```

**Comment by kbsali** [comment]
I'm having a somewhat similar issue : 
```
pg_dump -h db.PROJECT_REF.supabase.co -U postgres --clean --schema-only postgres > supabase_schema.sql2
pg_dump: error: query failed: ERROR:  permission denied for table identities
pg_dump: error: query was: LOCK TABLE auth.identities IN ACCESS SHARE MODE
```
@soedirgo the `GRANT` command you proposed did not fix the issue... any hint appreciated! 🙏

**Reply by bnjmnt4n** [reply]
I think the Supabase team just ran a migration creating that table, which might be why the table is locked. Maybe you can try again in a bit?

**Reply by soedirgo** [reply]
Hey @kbsali, this is a known issue - can you try running the following in the SQL editor?
```sql
grant all on auth.identities to postgres, dashboard_user;
```

**Reply by kbsali** [reply]
perfect, it works! thanks a lot @soedirgo !

**Reply by bokub** [reply]
@soedirgo Sorry for commenting in this old discusstion, but it's not working anymore.

I started encountering this error a few days ago:

```
pg_dump: error: query failed: ERROR:  permission denied for table schema_migrations
pg_dump: detail: Query was: LOCK TABLE "auth"."schema_migrations" IN ACCESS SHARE MODE
```

So I tried to run the following in my sql editor:

```sql
grant all on auth.schema_migrations to postgres, dashboard_user;
```

But the result is the following error:

```
ERROR:  42501: permission denied for table schema_migrations
```

Any idea how to solve this? Thanks

**Reply by soedirgo** [reply]
@bokub can you file a support ticket and include a link to this thread? It might be specific to your Supabase project.

**Reply by celine-m-s** [reply]
Hello! I face the same problem. Did you receive an answer from the support team?

**Reply by bokub** [reply]
I didn't take the time to write a ticket yet, but I should.

Please tell me if you receive an answer!

**Reply by celine-m-s** [reply]
Hey @bokub ! Their support team is super efficient. I received an answer in a couple of days. It was something very specific to our configuration so I don't think it would be relevant to post it here. I invite you to contact them directly.
Have a nice day!

**Reply by bokub** [reply]
I did contact them 11 days ago (oct. 16), but still waiting :crossed_fingers: 

Thanks anyway!

**Reply by bokub** [reply]
Update: I never received any answer from the support, but it looks like they solved my issue around Nov. 5 !

**Comment by Nditah** [comment]
I ran the command 
```sql 
grant all on auth.identities to postgres, dashboard_user;
```
Yet I still get 

```bash
pg_dump: error: query failed: ERROR:  permission denied for table schema_migrations
pg_dump: error: query was: LOCK TABLE realtime.schema_migrations IN ACCESS SHARE MODE
```

**Reply by dccabs** [reply]
**Disclaimer, i don't know much about DB's so follow this advice at your own risk.**

I had the same problem, and I ran the following 

```grant all on realtime.schema_migrations to postgres, dashboard_user;```

then 

```grant all on realtime.subscription to postgres, dashboard_user;```

then 

```grant all on realtime.subscription_id_seq to postgres, dashboard_user;```

basically keep checking the error log and replace `auth.identities` from your sql command above with the table that is throwing the error.

Eventually I got it to work doing this

**Comment by ajsharp** [comment]
I still get errors related to graphql stuff, even when running `pg_dump` with `--exclude-schema=graphql`.

```
pg_dump: error: query failed: ERROR:  permission denied for table _type
pg_dump: error: query was: LOCK TABLE graphql._type IN ACCESS SHARE MODE
```

When I try to grant privs to graphql tables, I get an error:
```
postgres=> GRANT ALL ON ALL SEQUENCES IN SCHEMA graphql TO postgres;
WARNING:  no privileges were granted for "seq_schema_version"
ERROR:  permission denied for sequence _type_id_seq
```

**Comment by jmarks-joshua** [comment]
To get this working I had to disable the PG_GRAPHQL extension, i tried all the grant permission commands given in this thread and none worked. I'm sure disabling the extension isn't ideal for most people that are actually using graphql, i thought about using it and had so many problems that I decided the standard REST api would be fine.

**Comment by Anj-interactivated** [comment]
```
grant all on _realtime.schema_migrations to postgres, dashboard_user;
grant all on _realtime.tenants to postgres, dashboard_user;
grant all on _realtime.extensions to postgres, dashboard_user;
```
this worked for me if someone is still experiencing the issue

**Comment by jdgamble555** [comment]
@soedirgo - Any updates on this? Changing permissions can obviously have side effects when pgadmin has no issues here.

J

**Comment by jwright04** [comment]
My error is slightly different.  I'm seeing

```
pg_dump: error: query failed: ERROR:  permission denied for table hooks
pg_dump: detail: Query was: LOCK TABLE "supabase_functions"."hooks" IN ACCESS SHARE MODE
```

Is there something I can do for this?

**Comment by semics-tech** [comment]
I also got similar errors. As I only wanted to migrate tables from the `public` schema, I limited the `pg_dump` command to export the `public` schema only using the `-n public` flag and it exported & imported without issues. Hope this helps someone.

**Reply by S3B4S** [reply]
This worked for me! Thanks @semics-tech ❤️

**Reply by filippobrigati** [reply]
Also for me, thanks
